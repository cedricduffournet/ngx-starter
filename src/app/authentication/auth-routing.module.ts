import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
