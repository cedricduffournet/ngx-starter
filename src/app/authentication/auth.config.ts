export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
