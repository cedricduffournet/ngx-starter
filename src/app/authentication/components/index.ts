export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
