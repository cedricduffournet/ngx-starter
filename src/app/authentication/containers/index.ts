export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
