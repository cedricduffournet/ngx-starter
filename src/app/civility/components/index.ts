export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
