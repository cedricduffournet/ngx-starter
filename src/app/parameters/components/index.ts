export * from './referenced-table-menu/referenced-table-menu.component';
