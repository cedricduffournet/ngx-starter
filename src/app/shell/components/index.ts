export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
