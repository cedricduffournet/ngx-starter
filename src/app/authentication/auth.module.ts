import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
