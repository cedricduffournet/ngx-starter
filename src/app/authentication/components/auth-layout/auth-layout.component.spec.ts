import { Component } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { AuthLayoutComponent } from '@app/authentication/components';

@Component({ selector: 'app-lang-switcher', template: '' })
class LangSwitcherStubComponent {}

@Component({ selector: 'app-logo', template: '' })
class LogoStubComponent {}

describe('AuthLayout', () => {
  let fixture: ComponentFixture<AuthLayoutComponent>;
  let component: AuthLayoutComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        AuthLayoutComponent,
        LangSwitcherStubComponent,
        LogoStubComponent
      ]
    });

    fixture = TestBed.createComponent(AuthLayoutComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display title 1', () => {
    component.title = 'test';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display title2', () => {
    component.title2 = 'title 2';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
