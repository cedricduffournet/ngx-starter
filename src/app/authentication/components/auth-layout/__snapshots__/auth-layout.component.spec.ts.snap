// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout should be created 1`] = `
<app-auth-layout>
  <div
    class="row"
  >
    <div
      class="col-md-12"
    >
      <div
        class="float-right pr-3 pt-3"
      >
        <app-lang-switcher />
      </div>
    </div>
  </div><div
    class="center-div login mt-5"
  >
    <span
      class="h2"
    >
      
    </span>
    
    <br />
    <section
      class="mt-3 card"
    >
      <div
        class="card-body"
      >
        <div
          class="text-center mb-3"
        >
          <a
            class="logo-link"
            href="#"
          >
            <app-logo
              color="black"
            />
          </a>
        </div>
      </div>
    </section>
  </div>
</app-auth-layout>
`;

exports[`AuthLayout should display title 1 1`] = `
<app-auth-layout
  title={[Function String]}
>
  <div
    class="row"
  >
    <div
      class="col-md-12"
    >
      <div
        class="float-right pr-3 pt-3"
      >
        <app-lang-switcher />
      </div>
    </div>
  </div><div
    class="center-div login mt-5"
  >
    <span
      class="h2"
    >
      test
    </span>
    
    <br />
    <section
      class="mt-3 card"
    >
      <div
        class="card-body"
      >
        <div
          class="text-center mb-3"
        >
          <a
            class="logo-link"
            href="#"
          >
            <app-logo
              color="black"
            />
          </a>
        </div>
      </div>
    </section>
  </div>
</app-auth-layout>
`;

exports[`AuthLayout should display title2 1`] = `
<app-auth-layout
  title2={[Function String]}
>
  <div
    class="row"
  >
    <div
      class="col-md-12"
    >
      <div
        class="float-right pr-3 pt-3"
      >
        <app-lang-switcher />
      </div>
    </div>
  </div><div
    class="center-div login mt-5"
  >
    <span
      class="h2"
    >
      
    </span>
    
    <span
      class="h2 text-black-50"
    >
      <br />
       title 2 
    </span>
    <br />
    <section
      class="mt-3 card"
    >
      <div
        class="card-body"
      >
        <div
          class="text-center mb-3"
        >
          <a
            class="logo-link"
            href="#"
          >
            <app-logo
              color="black"
            />
          </a>
        </div>
      </div>
    </section>
  </div>
</app-auth-layout>
`;
